### DISCUSSION TITLE
error during manual install: ${USE_OLLAMA_DOCKER,,}: bad substitution

### DISCUSSION THREAD

**Discussion by arielelkin** [open-webui/open-webui#3269] [General]
Trying to install Open WebUI manually following [the official instructions](https://docs.openwebui.com/getting-started/#build-and-install-%EF%B8%8F)

the `pip install` and `bash start.sh` commands yield the following errors 

```
Loading WEBUI_SECRET_KEY from file, not provided as an environment variable.
Loading WEBUI_SECRET_KEY from .webui_secret_key
start.sh: line 23: ${USE_OLLAMA_DOCKER,,}: bad substitution
start.sh: line 25: ${USE_CUDA_DOCKER,,}: bad substitution
start.sh: line 52: exec: uvicorn: not found
```

I'm on macOS 14.4
python --version
Python 3.12.4
node --version
v22.3.0

Any thoughts?

**Comment by jlala** [comment]
Hello, do you have a solution?

**Reply by arielelkin** [reply]
Nope.

**Comment by arielelkin** [comment]
@tjbck Any thoughts?

**Comment by pascal-beaulne** [comment]
Have you tried using rye sync and rye run open-webui dev?
It can also be an issue with your Bash version.

**Reply by arielelkin** [reply]
Ah, rye solved it. Thanks @pascal-beaulne 

I'm using zsh 5.9, so I don't think the shell is the issue. This was Python dependency hell.

**Reply by sgkassar** [reply]
HI @arielelkin how did you fix this dependency? ty

**Reply by arielelkin** [reply]
@sgkassar as @pascal-beaulne suggested: installing rye, calling `rye sync‘ and then `rye run open-webui serve`

**Reply by mirkonasato** [reply]
Just encountered the same issue, and it was in fact because of the bash version. macOS by default includes v3. The error went away after installing bash v5 with brew.
